src/ImageClassifier/index.js

```diff
@@ -0,0 +1,263 @@
+    const imageResize = [IMAGE_SIZE, IMAGE_SIZE];
+    if (this.modelName == "doodlenet" || this.modelUrl) imgToPredict = imgToTensor(imgToPredict, imageResize);
+
+    if (this.modelUrl) {
+      await tf.nextFrame();
+      const predictedClasses = tf.tidy(() => {
+        const predictions = this.model.predict(imgToPredict);
+        return Array.from(predictions.as1D().dataSync());
+      });
+
+      const results = await predictedClasses
+        .map((confidence, index) => {
+          const label =
+            this.mapStringToIndex.length > 0 && this.mapStringToIndex[index]
+              ? this.mapStringToIndex[index]
+              : index;
+          return {
+            label,
+            confidence,
+          };
+        })
+        .sort((a, b) => b.confidence - a.confidence);
+      return results;
+    }
+
+    const results = this.model
+      .classify(imgToPredict, numberOfClasses)
+      .then(classes => classes.map(c => ({ label: c.className, confidence: c.probability })));
+
+    return results;
+  }
+
+  /**
+   * Classifies the given input and takes a callback to handle the results
+   * @param {HTMLImageElement | HTMLCanvasElement | object | function | number} inputNumOrCallback -
+   *    takes any of the following params
+   * @param {HTMLImageElement | HTMLCanvasElement | object | function | number} numOrCallback -
+   *    takes any of the following params
+   * @param {function} cb - a callback function that handles the results of the function.
+   * @return {function} a promise or the results of a given callback, cb.
+   */
+  async classify(inputNumOrCallback, numOrCallback, cb) {
+    const { image, number, callback } = handleArguments(inputNumOrCallback, numOrCallback, cb)
+      .require('image',
+        "No input image provided. If you want to classify a video, use classifyStart."
+      );
+    return callCallback(this.classifyInternal(image, number), callback);
+  }
+
+    /**
+   * Continuously classifies each frame of the given input
+   * @param {HTMLVideoElement | object | function | number} inputNumOrCallback -
+   *    takes any of the following params
+   * @param {object | function | number} numOrCallback - a number of labels to return for the image classification.
+   * @param {function} cb - a callback function that handles the results of the function.
+   * @return {function} a promise or the results of a given callback, cb.
+   */
+    async classifyStart(inputNumOrCallback, numOrCallback, cb) {
+    const { image, number, callback } = handleArguments(inputNumOrCallback, numOrCallback, cb)
+      .require('image', "No input provided.");
+
+    // Function to classify a single frame
+    const classifyFrame = async () => {
+      await mediaReady(image, true);
+      await this.classifyInternal(image, number);
+      // call the callback function
+      callCallback(this.classifyInternal(image, number), callback);
```

Comment:
This will call the `this.classifyInternal` function twice.  The `callCallback` function returns the underlying promise so you can `await` it.  You don't need line 209.  You can just do `await callCallback(this.classifyInternal(image, number), callback);`

Reply:
Noted, thanks for catching this, will be corrected in the upcoming Image Classifier update